File: CreationClasses/metier/personne.py
from abc import ABC


class Personne (ABC):

    def __init__(self, nom = "Inconnu") -> None:
        self.__nom = nom

    def get_nom(self) -> str:
        return self.__nom

    def set_nom(self, nom: str) -> None :
        self.__nom = nom

    def __eq__(self, o: object) -> bool:
        return self.__nom == o.__nom

    def __hash__(self) -> int:
        return self.__nom.__hash__()

    def afficher(self) -> str:
        return self.__nom

File: CreationClasses-Java/src/domaine/Personne.java
package domaine;

import java.util.Objects;

public abstract class Personne {

    private String nom;

    public Personne() {
        nom = "Inconnu";
    }

    public Personne(String nom) {
        this.nom = nom;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    @Override
    public boolean equals(Object o) {
        Personne other = (Personne)o;
        return other.nom.equals(nom);
    }

    @Override
    public int hashCode() {
        return Objects.hash(nom);
    }

    // On devrait utiliser toString, mais j'ai utilisé "afficher" pour l'exemple
    public abstract String afficher();
}
